autopilot/hostfilter.go

```diff
@@ -44,6 +44,26 @@ var (
 	errContractExpired           = errors.New("contract has expired")
 )
 
+func hostErrCounts(errs []error, counts map[string]int) {
+	list := map[string]error{
```

Comment:
(nit) I think I'd rather have a struct with fields for each type of error that can happen including an "unknown" category just in case that is created in `isUsableHost` and has a `Merge` method so that we can combine them until we have a final one. 

e.g. `UnusableHostResults` and it can be created from a slice of errors in `isUsableHost`.

Reply:
I'm not a big fan but I'm sure I'll appreciate it tomorrow.